Libplanet/Store/StateStoreExtensions.cs

```diff
@@ -0,0 +1,87 @@
+#nullable enable
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Security.Cryptography;
+using System.Text;
+using Bencodex.Types;
+using Libplanet.Store.Trie;
+
+namespace Libplanet.Store
+{
+    /// <summary>
+    /// Convenient extension methods for <see cref="IStateStore"/>.
+    /// </summary>
+    public static class StateStoreExtensions
+    {
+        /// <summary>
+        /// The internal bytes encoding of raw state keys.
+        /// </summary>
+        internal static readonly Encoding KeyEncoding = Encoding.UTF8;
+
+        /// <summary>
+        /// Records <paramref name="rawStatesDelta"/> which is based on the previous state
+        /// root, and returns the new state root.
+        /// </summary>
+        /// <param name="stateStore">The <see cref="IStateStore"/> to records
+        /// the <paramref name="rawStatesDelta"/>.</param>
+        /// <param name="previousStateRootHash">The state root hash on which
+        /// the <paramref name="rawStatesDelta"/> is based.</param>
+        /// <param name="rawStatesDelta">The raw states delta to be recorded.</param>
+        /// <param name="rehearsal">If turned on, the <paramref name="rawStatesDelta"/> is not
+        /// recorded to the <paramref name="stateStore"/>, but only shadow trie root is returned.
+        /// Turned off by default.</param>
+        /// <returns>The new state root hash.</returns>
```

Comment:
I'm not that sure it can treat the trie as `state root hash` (feel like `Hash`) 🤔

Reply:
Amended: c3548d94ba1190d03ac697092fa56ed1ffa3038a.